webapp/IronCalc/src/components/WorkbookSettings/WorkbookSettingsDialog.tsx

```diff
@@ -0,0 +1,429 @@
+import styled from "@emotion/styled";
+import {
+  Autocomplete,
+  Box,
+  Dialog,
+  FormControl,
+  MenuItem,
+  Select,
+  TextField,
+} from "@mui/material";
+import { Check, X } from "lucide-react";
+import { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { theme } from "../../theme";
+
+type WorkbookSettingsDialogProps = {
+  open: boolean;
+  onClose: () => void;
+  initialLocale: string;
+  initialTimezone: string;
+  onSave?: (locale: string, timezone: string) => void;
+};
+
+const WorkbookSettingsDialog = (properties: WorkbookSettingsDialogProps) => {
+  const { t } = useTranslation();
+  const locales = ["en-US", "en-GB", "de-DE", "fr-FR", "es-ES"];
+  const timezones = [
+    "Berlin, Germany (GMT+1)",
+    "New York, USA (GMT-5)",
+    "Tokyo, Japan (GMT+9)",
+    "London, UK (GMT+0)",
+    "Sydney, Australia (GMT+10)",
+  ];
+  const [selectedLocale, setSelectedLocale] = useState<string>(
+    properties.initialLocale && locales.includes(properties.initialLocale)
+      ? properties.initialLocale
+      : locales[0],
+  );
+  const [selectedTimezone, setSelectedTimezone] = useState<string>(
+    properties.initialTimezone && timezones.includes(properties.initialTimezone)
+      ? properties.initialTimezone
+      : timezones[0],
+  );
+
+  const handleSave = () => {
+    if (properties.onSave && selectedLocale && selectedTimezone) {
+      properties.onSave(selectedLocale, selectedTimezone);
+    }
+    properties.onClose();
+  };
+
+  // Ensure selectedLocale is always a valid locale
+  const validSelectedLocale =
+    selectedLocale && locales.includes(selectedLocale)
+      ? selectedLocale
+      : locales[0];
+
+  // Ensure selectedTimezone is always a valid timezone
+  const validSelectedTimezone =
+    selectedTimezone && timezones.includes(selectedTimezone)
+      ? selectedTimezone
+      : timezones[0];
+
```

Comment:
There is a subtle error here, that happens a lot with React. The state of the Dialog persist across opens.

1. Open dialog with a workbook with fr locale. All good
2. Switch to a workbook with de locale. When you open the dialog is still with the fr locale!!!


To solve this you need to set the locale each time the initLocale changes:
```ts
useEffect(() => {
  if (properties.open) {
    setSelectedLocale(properties.initialLocale);
    setSelectedTimezone(properties.initialTimezone);
    setSelectedLanguage(properties.initialLanguage);
  }
}, [
  properties.open,
  properties.initialLocale,
  properties.initialTimezone,
  properties.initialLanguage,
]);
```